src/components/data-management/SamplesTable.jsx

```diff
@@ -221,19 +235,14 @@ const SamplesTable = forwardRef((props, ref) => {
   };
 
   useEffect(() => {
-    if (!samples[activeProject.samples[0]]) {
+    if (activeProject.samples.length === 0) {
       setTableData([]);
       return;
     }
 
-    // Set table data
     const newData = activeProject.samples.map((sampleUuid, idx) => {
-      // upload problems sometimes lead to partial updates and incosistent states
-      // in this situation it's possible that the sampleUuid does not exist
-      // this a temporary fix so that the whole UI doesn't crash preventing the
-      // user from removing the dataset or uploading another one.
-      // this situation should be included into the SamplesTable.jsx tests
-      const sampleFiles = samples[sampleUuid]?.files || {};
+      const sampleData = samples[sampleUuid] || sampleUploadErrorTemplate;
+      const sampleFiles = sampleData.files;
```

Comment:
if `sampleData` is equal to `sampleUploadErrorTemplate`, there will be no `files` in it. There will be `sampleFiles` instead

Reply:
yes, my bad. Fixed.